sdk/src/wallet/operations/transaction/build_transaction.rs

```diff
@@ -35,100 +31,28 @@ where
         let protocol_parameters = self.client().get_protocol_parameters().await?;
 
         let mut inputs: Vec<Input> = Vec::new();
-        let mut context_inputs = HashSet::new();
-
-        let issuance = self.client().get_issuance().await?;
-        let latest_slot_commitment_id = issuance.latest_commitment.id();
-
-        let mut needs_commitment_context = false;
-
-        for (idx, input) in selected_transaction_data.inputs.iter().enumerate() {
-            // Transitioning an issuer account requires a BlockIssuanceCreditContextInput.
-            if let Output::Account(account) = &input.output {
-                if account.features().block_issuer().is_some() {
-                    context_inputs.insert(ContextInput::from(BlockIssuanceCreditContextInput::from(
-                        account.account_id_non_null(input.output_id()),
-                    )));
-                }
-            }
-
-            // Inputs with timelock or expiration unlock condition require a CommitmentContextInput
-            if input
-                .output
-                .unlock_conditions()
-                .map_or(false, |u| u.iter().any(|u| u.is_timelock() || u.is_expiration()))
-            {
-                needs_commitment_context = true;
-            }
 
+        for input in &selected_transaction_data.inputs {
             inputs.push(Input::Utxo(UtxoInput::from(*input.output_id())));
-
-            if selected_transaction_data.mana_rewards.get(input.output_id()).is_some() {
-                context_inputs.insert(ContextInput::from(RewardContextInput::new(idx as _)?));
-                needs_commitment_context = true;
-            }
-        }
-
-        // TODO https://github.com/iotaledger/iota-sdk/issues/1937
-        for output in selected_transaction_data
-            .outputs
-            .iter_mut()
-            .filter(|o| o.is_delegation())
-        {
-            // Created delegations have their start epoch set, and delayed delegations have their end set
-            if output.as_delegation().delegation_id().is_null() {
-                *output = DelegationOutputBuilder::from(output.as_delegation())
-                    .with_start_epoch(protocol_parameters.delegation_start_epoch(latest_slot_commitment_id))
-                    .finish_output()?;
-            } else {
-                *output = DelegationOutputBuilder::from(output.as_delegation())
-                    .with_end_epoch(protocol_parameters.delegation_end_epoch(latest_slot_commitment_id))
-                    .finish_output()?;
-            }
-            needs_commitment_context = true;
         }
 
         // Build transaction
 
-        // TODO: Add an appropriate mana allotment here for this account
         let mut builder = Transaction::builder(protocol_parameters.network_id())
```

Comment:
I am just seeing now that this isn't using `with_creation_slot` which means the default will use system time that isn't verified against the node. We should preferably set it to `self.client().get_slot_index()`

Reply:
but that's what gets called?

Reply:
Hmm doesn't look like it

Reply:
oh, weird, I thought I changed that or something